addons/l10n_ec/models/account_move.py

```diff
@@ -0,0 +1,259 @@
+            ("auto_printer", "Auto Printer"),
+            ("electronic", "Electronic"),
+        ],
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default="electronic",
+    )
+
+    def _get_l10n_ec_identification_type(self):
+        self.ensure_one()
+        move = self
+        _valid, l10n_ec_type_sri = move.partner_id.commercial_partner_id.check_vat_ec()
+        l10n_ec_is_exportation = move.partner_id.commercial_partner_id.country_id.code != 'EC'
+        identification_code = False
+        if move.move_type in ("in_invoice", "in_refund"):
+            if l10n_ec_type_sri == "Ruc":
+                identification_code = "01"
+            elif l10n_ec_type_sri == "Cedula":
+                identification_code = "02"
+            else:
+                identification_code = "03"
+        elif move.move_type in ("out_invoice", "out_refund"):
+            if not l10n_ec_is_exportation:
+                if l10n_ec_type_sri == "Ruc":
+                    identification_code = "04"
+                elif l10n_ec_type_sri == "Cedula":
+                    identification_code = "05"
+                elif l10n_ec_type_sri == "Pasaporte":
+                    identification_code = "06"
+                elif l10n_ec_type_sri == "Consumidor":
+                    identification_code = "07"
+            else:
+                if l10n_ec_type_sri == "Ruc":
+                    identification_code = "20"
+                elif l10n_ec_type_sri == "Pasaporte":
+                    identification_code = "21"
+                else:
+                    identification_code = "09"
+        return identification_code
+
+    @api.model
+    def _get_l10n_ec_documents_allowed(self, identification_code):
+        documents_allowed = self.env['l10n_latam.document.type'].browse()
+        for document_ref in _DOCUMENTS_MAPPING.get(identification_code, []):
+            document_allowed = self.env.ref('l10n_ec.%s' % document_ref, False)
+            if document_allowed:
+                documents_allowed |= document_allowed
+        return documents_allowed
+
+    def _get_l10n_latam_documents_domain(self):
+        if self.company_id.country_id.code != "EC":
+            return super(AccountMove, self)._get_l10n_latam_documents_domain()
+        internal_type = self.env.context.get("internal_type", False)
+        if internal_type == "purchase_liquidation":
+            internal_types = ["purchase_liquidation"]
+            domain = [
+                ("internal_type", "in", internal_types),
+                ("country_id", "=", self.company_id.country_id.id),
+            ]
+        else:
+            domain = super(AccountMove, self)._get_l10n_latam_documents_domain()
+        latam_type = internal_type or "invoice"
+        if self.move_type in ("out_refund", "in_refund"):
+            latam_type = "credit_note"
+        if self.debit_origin_id:
+            latam_type = "debit_note"
```

Comment:
Why redo this as it is probably already in the domain...  And you are appending it...

Reply:
The move_type variable of account.move does not have a value for debit_note, the way I determine that it is a debit note is by verifying that this field is full, that is why it is implemented with that condition, since I must also delimit the documents depending on the type of document, we can review it functionally

Reply:
Domain to be clear enough for others to understand